ScoreSaber/Core/Services/ScoreSaberRichPresenceService.cs

```diff
@@ -0,0 +1,448 @@
+                _socket.OnClose += OnCloseCallback;
+                _socket.Connect();
+                _log.Notice("Rich presence service initialized.");
+            } catch (Exception ex) {
+                _log.Error($"Failed to initialize socket: {ex.Message}");
+            }
+        }
+
+        public void ToggleRichPresence(bool enable) {
+            if (_isEnabled == enable) {
+                _log.Info("Rich presence toggle state unchanged.");
+                return;
+            }
+
+            _isEnabled = enable;
+
+            if (_isEnabled) {
+                _log.Info("Enabling rich presence service...");
+                Initialize();
+            } else {
+                _log.Info("Disabling rich presence service...");
+                DisconnectSocket();
+            }
+        }
+
+        private void DisconnectSocket() {
+            if (_socket != null) {
+                _socket.Disconnect();
+                _socket.OnOpen -= OnOpenCallback;
+                _socket.OnMessage -= OnMessageCallback;
+                _socket.OnClose -= OnCloseCallback;
+                _socket = null;
+                _userProfileChannel = null;
+                _log.Info("Socket disconnected. Service is now offline.");
+            }
+        }
+
+        private void SendMenuEvent() {
+            if (_playerService?.localPlayerInfo == null) {
+                _log.Warn("Local player info is null, unable to send rich presence data.");
+                return;
+            }
+
+            if (_socket == null) {
+                _log.Warn("Socket is null, unable to send rich presence data.");
+                return;
+            }
+
+            var jsonObject = new SceneChangeEvent {
+                Timestamp = TimeRightNow,
+                Scene = Scene.menu
+            };
+
+            SendUserProfileChannel("scene_change", jsonObject);
+        }
+
+        private void OnCloseCallback(ushort code, string message) {
+            _log.Warn($"Rich presence socket closed: {code} - {message}");
+            if (_isDisposed) return;
+
+            if (_retryAttempt >= MaxRetries) {
+                _log.Error("Max reconnection attempts reached. Unable to reconnect.");
+                Dispose();
+                return;
+            }
+
+            _retryAttempt++;
+            _log.Warn($"Attempting to reconnect to rich presence socket... Attempt {_retryAttempt}");
+            _socket?.Connect();
+        }
+
+        public void SendUserProfileChannel(string @event, object message, TimeSpan? timeSpan = null) {
+            if (_userProfileChannel == null) {
+                _log.Warn("User profile channel is null, unable to send message.");
+                return;
+            }
+
+            _userProfileChannel.Push(@event, message, timeSpan);
+        }
+
+        private void OnMessageCallback(Message message) {
+            //_log.Info($"Received message: {message.Payload}");
+        }
```

Comment:
Not really what I meant. Also if you are gonna use Log.Debug make the message more meaningful.

But I am worried this could spam the log a bit much.

Reply:
Removed logging messages entirely, there was nothing useful